views/overlays/admin.hbs

```diff
@@ -185,6 +211,48 @@
 		toastNotify();
 	}
 
+	function sendChallongeUpdate() {
+		var data = {
+			'challongeUrl': document.getElementById('challongeUrl').value,
+		};
+		challongeSocket.emit('update challonge', data);
+		toastNotify();
+	}
+
+	function createMatchList(data) {
+		var matchList = $('#challongeMatchList');
+		if (matchList) {
+			matchList.remove();
+		}
+
+		var selectList = document.createElement('select');
+		selectList.id = 'challongeMatchList';
+		document.getElementById('challongeMatchListPlaceholder').appendChild(selectList);
+
+		var defaultOption = document.createElement('option');
+		defaultOption.text = 'Select a match to display';
+		selectList.appendChild(defaultOption);
+
+		data.upcomingMatches.forEach(function(match){
+			var option = document.createElement('option');
+			var playerOne = data.players[match.match.player1_id];
+			var playerTwo = data.players[match.match.player2_id];
+			option.text = match.match.identifier+': '+playerOne+' vs. '+playerTwo;
+			option.value = playerOne+','+playerTwo;
```

Comment:
Could you make the option values be the match identifier and then fetch the player names in the `.change` function? It'll probably take some rethinking on the `upcomingMatches` data structure, but I think it's worth it to avoid doing the splits to parse the player names out.